cmd/kusk/cmd/upgrade.go

```diff
@@ -115,49 +125,54 @@ var upgradeCmd = &cobra.Command{
 
 				fmt.Println("✅ Envoy Fleets upgraded")
 			case "kusk-gateway-api":
-				fmt.Println("✅ kusk API server is already installed. Upgrading...")
-				if err := applyf(filepath.Join(dir, manifests_dir, "api_server.yaml")); err != nil {
-					kuskui.PrintError("❌ failed upgrading API Server", err.Error())
-					reportError(err)
-					return err
-				}
-				if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, "kusk-gateway-api", time.Duration(5*time.Minute), "instance"); err != nil {
-					kuskui.PrintError("❌ failed upgrading API Server", err.Error())
-					reportError(err)
-					return err
-				}
+				if !utils.IsUptodate(getVersions(deployment.Name, "kusk-gateway-api", deployment)) {
+					fmt.Println("✅ kusk API server is already installed. Upgrading...")
+					if err := applyf(filepath.Join(dir, manifests_dir, "api_server.yaml")); err != nil {
+						kuskui.PrintError("❌ failed upgrading API Server", err.Error())
+						reportError(err)
+						return err
+					}
+					if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, "kusk-gateway-api", time.Duration(5*time.Minute), "instance"); err != nil {
+						kuskui.PrintError("❌ failed upgrading API Server", err.Error())
+						reportError(err)
+						return err
+					}
 
-				if err := applyf(filepath.Join(dir, manifests_dir, "api_server_api.yaml")); err != nil {
-					kuskui.PrintError("❌ failed upgrading API Server", err.Error())
-					reportError(err)
-					return err
+					if err := applyf(filepath.Join(dir, manifests_dir, "api_server_api.yaml")); err != nil {
+						kuskui.PrintError("❌ failed upgrading API Server", err.Error())
+						reportError(err)
+						return err
+					}
+					kuskui.PrintStart("API Server installed")
 				}
-				kuskui.PrintStart("API Server installed")
 			case "kusk-gateway-dashboard":
-				fmt.Println("✅ kusk Dashboard is already installed. Upgrading...")
-				if err := applyf(filepath.Join(dir, manifests_dir, "dashboard.yaml")); err != nil {
-					kuskui.PrintError("❌ failed upgrading Dashboard", err.Error())
-					reportError(err)
-					return err
-				}
-				if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, "kusk-gateway-dashboard", time.Duration(5*time.Minute), "instance"); err != nil {
-					kuskui.PrintError("❌ failed upgrading Dashboard", err.Error())
-					reportError(err)
-					return err
-				}
+				if !utils.IsUptodate(getVersions("kusk-gateway-dashboard", "kusk-gateway-dashboard", deployment)) {
```

Comment:
Shouldn't these be constants? `"kusk-gateway-dashboard"` is repeated in the line above in the `case` statement.